store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -6147,10 +6152,21 @@ private void checkPasswordStrength(String password, Account acct, Cos cos, ZMuta
                     new Argument(Provisioning.A_zimbraPasswordMaxLength, maxLength, Argument.Type.NUM));
         }
 
-        if (LC.allow_username_within_password.booleanValue()
-                && StringUtils.containsIgnoreCase(password, acct.getUCUsername())) {
-            throw AccountServiceException.INVALID_PASSWORD("password contains username",
-                    new Argument("zimbraPasswordAllowUsername", "", Argument.Type.STR));
+        // userName is passed from createAccount() and else condition is invoked 
+        // when called from other Api's Changepassword and Resetpassword Api
+        if (userName != null) {
+            if (LC.allow_username_within_password.booleanValue()
```

Comment:
@gabhinavTI change the default value of LC allow_username_within_password to true and do the check in the code accordingly. So if its true there is not check done for username present in password. If false then only do the check.